### DISCUSSION TITLE
Add an option to control setup_future_usage Stripe setting

### DISCUSSION THREAD

**Discussion by david-lobo** [duncanmcclean/simple-commerce#929] [Help] [answered]
By default when using Stripe as the payment gateway, the Stripe API setting setup_future_usage is set to 'off_session'.  This results in the customer getting the following message on the Checkout - Payment page which is rendered by Stripe.

"By providing your card information, you allow COMPANY_NAME to charge your card for future payments in accordance with their terms."

Could we please have the ability to either omit this setting or set setup_future_usage to 'on_session'? 

This would allow this message to not be shown, which is preferable for use cases where we do not intend to use the payment card in future.

**Comment by duncanmcclean** [comment]
Hello 👋

You may override the values Simple Commerce uses to create the Stripe Payment Intent by providing a `payment_intent_callback` closure in your Stripe gateway's config, like this:

```php
// config/simple-commerce.php

\DoubleThreeDigital\SimpleCommerce\Gateways\Builtin\StripeGateway::class => [
    'key' => env('STRIPE_KEY'),
    'secret' => env('STRIPE_SECRET'),
    'payment_intent_data' => function (ContractsOrder $order) {
        return [
            'setup_future_usage' => 'on_session',
        ];
    },
],
```

**Note:** You won't be able to use `php artisan config:cache` when doing this (although, arguably, that doesn't make much of a performance difference anyway).

**Reply by david-lobo** [reply, marked as answer]
Hi Duncan, Thanks verry much for the quick response.  Yes that suggestion works well and i've now managed to remove the text.  In case it helps anyone else, the setting needs to be 'null' in order for the text to be removed.

```
// config/simple-commerce.php

\DoubleThreeDigital\SimpleCommerce\Gateways\Builtin\StripeGateway::class => [
    'key' => env('STRIPE_KEY'),
    'secret' => env('STRIPE_SECRET'),
    'payment_intent_data' => function (ContractsOrder $order) {
        return [
            'setup_future_usage' => null,
        ];
    },
],
```

**Reply by duncanmcclean** [reply]
Gotcha - apologies, must have used the wrong example in the array. Glad you've got it working 🙌
